src/main/java/gov/usgs/owi/nldi/controllers/BaseController.java

```diff
@@ -58,38 +58,53 @@ public BaseController(LookupDao inLookupDao, StreamingDao inStreamingDao, Naviga
 
 	protected void streamFlowLines(HttpServletResponse response,
 			String comid, String navigationMode, String stopComid, String distance, boolean legacy) throws Exception {
+
+		String featureType;
 		Map<String, Object> parameterMap = parameters.processParameters(comid, navigationMode, distance, stopComid);
-		FlowLineTransformer transformer = new FlowLineTransformer(response);
+
 		if (legacy) {
 			String sessionId = getSessionId(parameterMap, response);
 			if (null != sessionId) {
 				parameterMap.put(StreamingDao.SESSION_ID, sessionId);
-				addContentHeader(response);
-				streamResults(transformer, BaseDao.FLOW_LINES_LEGACY, parameterMap);
+				featureType = BaseDao.FLOW_LINES_LEGACY;
+			} else {
+				return;
```

Comment:
Aren't multiple exit points in a method evil?

Reply:
'evil'?  No.
A compromise?  Yes.